guillotina_authentication/api.py

```diff
@@ -126,6 +130,10 @@
         else:
             callback = request.url.query['callback']
 
+        forwarded_proto = request.headers.get('X-Forwarded-Proto', None)
+        if forwarded_proto and forwarded_proto != request.scheme:
+            callback = callback.replace(request.scheme, forwarded_proto)
```

Comment:
This will replace any values of 'http' in a url. For example, 'www.httpfoobar.com' -> 'www.httpsfoobar.com'.

should be: `callback.replace(request.scheme '://', forwarded_proto + '://')`